engine/classes/Elgg/Di/ServiceProvider.php

```diff
@@ -242,6 +244,10 @@ public function __construct(\Elgg\AutoloadManager $autoload_manager) {
 		});
 
 		$this->setClassName('widgets', '\Elgg\WidgetsService');
+		
+		$this->setFactory('uploads', function(ServiceProvider $c) {
+			return $c->request->files;
```

Comment:
Right now we kinda can't (that's why I made #8238). We can only make a global function that returns it, like elgg_get_session().